### PR TITLE
[BUGFIX] Hide applications belonging to deactivated jobs

### PR DESCRIPTION
**Pull request by pm-sschreiber**
As the deactivated-flag is meant as a general on-off switch for jobs and all related applications, these applications should be filtered out.

### CODE REVIEW

File: Classes/Domain/Repository/AjaxApplicationRepository.php [hunk 1]
```diff
@@ -25,8 +25,22 @@ public function getTotalResultsOfQuery(ApplicationQuery $query)
         /** @var QueryBuilder */
         $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_ats_domain_model_application');
 
-        $count = $queryBuilder->count('uid')
-            ->from('tx_ats_domain_model_application')
+        $queryBuilder->getRestrictions()
+            ->removeByType(StartTimeRestriction::class)
+            ->removeByType(EndTimeRestriction::class)
+            ->removeByType(HiddenRestriction::class);
```
Comment (mbrodala) [inline Classes/Domain/Repository/AjaxApplicationRepository.php]: Why are these removed?

Comment (pm-sschreiber) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Because of the join to the jobs table. With applications, these fields are not used anyway, but jobs may be hidden or limited for frontend access via start/endtime. Since we are in backend context, these jobs (and connected applications) still need to show up.

File: Classes/Domain/Repository/AjaxApplicationRepository.php [hunk 2]
```diff
@@ -40,18 +54,32 @@ public function findWithQuery(ApplicationQuery $query)
...
-            'tstamp',
-            'firstname',
-            'surname',
-            'job',
-            'status',
-            'rating',
-            'disability',
-            'employed'
-        )->from('tx_ats_domain_model_application')
+            'application.uid AS uid',
```
[the remark this answers is not in the document]
Comment (pm-sschreiber) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Sure? Doctrine kept complaining when I left some of the AS statements out.

File: Classes/Domain/Repository/AjaxApplicationRepository.php (hunk 1 above)
Comment (mbrodala) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Still sounds like this should be done separately then.

File: Classes/Domain/Repository/AjaxApplicationRepository.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbrodala) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Yes sure. Maybe you mean something like `'uid'` which would be ambiguous while `<table>.uid` is unique.

File: Classes/Domain/Repository/AjaxApplicationRepository.php (hunk 1 above)
Comment (pm-sschreiber) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Where exactly should it be done?

File: Classes/Domain/Repository/AjaxApplicationRepository.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (pm-sschreiber) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: Ah, got it. Preparing a fix for that.

File: Classes/Domain/Repository/AjaxApplicationRepository.php (hunk 1 above)
Comment (mbrodala) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: In a separate PR/commit AFAIS.

Comment (mbrodala) [reply Classes/Domain/Repository/AjaxApplicationRepository.php]: As discussed I get the intention now, so please add a small comment here and below to quickly explain this. Otherwise the next one who looks at this will have the same question. Possibly me. ;-)
